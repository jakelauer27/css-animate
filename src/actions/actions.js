export const loadAnimation = (animation) => ({
  type: 'LOAD_ANIMATION',
  animation
})
