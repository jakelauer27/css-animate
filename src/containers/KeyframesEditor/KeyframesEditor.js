import React, { Component } from 'react'
import * as formValidation from '../../utils/formValidators'
import { connect } from 'react-redux'
import { loadAnimation } from '../../actions/actions'
import { updateKeyframes } from '../../utils/keyframesInsertion'

class KeyframesEditor extends Component {

  saveKeyframesStages(e) {
    formValidation.keyframeStage(e.target, e.target.value)
    document.querySelector('.stop-btn').click()
    let newAnimation = JSON.stringify(JSON.parse(this.props.animation))
    let stageLabel = e.target.classList[1]
    let stageIndex = ''
    newAnimation.keyframes.sections.forEach( (stage, i) => {
       if (stage.name === stageLabel) {
         stageIndex = i
       }
    })
    newAnimation.keyframes.sections[stageIndex].label = e.target.value || '%'
    this.props.updateKeyframesStages(stageIndex, e.target.value, newAnimation)
    updateKeyframes(newAnimation.keyframes)
  }

  saveKeyframesProps(e) {
    formValidation.keyframeValue(e.target, e.target.value)
    document.querySelector('.stop-btn').click()
    let newAnimation = {...this.props.animation}
    let stageLabel = e.target.parentElement.parentElement.childNodes[0].classList[1]
    let propLabel = e.target.classList[1]
    let stageIndex = ''
    let propIndex = ''
    newAnimation.keyframes.sections.forEach( (stage, i) => {
       if (stage.name === stageLabel) {
         stageIndex = i
       }
    })
    newAnimation.keyframes.sections[stageIndex].properties.forEach( (prop, i) => {
       if (prop.name === propLabel) {
         propIndex = i
       }
    })
    newAnimation.keyframes.sections[stageIndex].properties[propIndex].value = e.target.value
    this.props.updateAnimation(newAnimation)
    updateKeyframes(newAnimation.keyframes)
  }

  render() {
    const { animation } = this.props
    if (!animation.keyframes) {
      return <div></div>
    }
    return (
      <div className='keyframes-container'>
        <div className='keyframe'>
          <p className='keyframes-name'><span>@keyframes </span>{animation.keyframes.name} <span> {'{'}</span></p>
          {
            animation.keyframes.sections.map( (section, i) => {
              return (
                <div className='keyframes-section' key={i}>
                  <input className={`keyframes-label ${section.name}`} 
                    value={section.label} 
                    type='text'
                    onChange={e => this.saveKeyframesStages(e)}/> 
                    <span> {'{'}</span>
                  {
                    section.properties.map( (prop, i) => {
                      return (
                        <div className='props-container' key={i}>
                          <p className='keyframe-property'>{prop.name}<span>:</span></p>
                          <input className={`keyframe-prop-value ${prop.name}`} 
                            type='text' 
                            value={prop.value}
                            onChange={e => this.saveKeyframesProps(e)}></input>
                        </div>                   
                        )
                      })
                    }
                  <p className='close-curly'>{'}'}</p>
                </div>
              )
            }) 
          }
          <p className='close-curly'>{'}'}</p>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  animation: state.animation
})

export const mapDispatchToProps = (dispatch) => ({
  updateAnimation: (animation) => dispatch(loadAnimation(animation))
})

export default connect(mapStateToProps, mapDispatchToProps)(KeyframesEditor)