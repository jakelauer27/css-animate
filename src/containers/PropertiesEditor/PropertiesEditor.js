import React, { Component } from 'react'
import { connect } from 'react-redux'
import { validateAnimationProp } from '../../utils/formValidators'
import { loadAnimation } from '../../actions/actions'
import { uid } from 'react-uid';


const aniPropsLabels = ['duration', 'timing-function', 'delay', 'iteration-count', 'direction', 'fill-mode']
const aniProps = ['duration', 'timingFunction', 'delay', 'iterationCount', 'direction', 'fillMode']

class PropertiesEditor extends Component {

  saveForm(e) {
    document.querySelector('.stop-btn').click()
    validateAnimationProp(e.target, e.target.value)
    let newAnimation = JSON.parse(JSON.stringify(this.props.animation))
    newAnimation.properties[e.target.classList[1]] = e.target.value
    this.props.updateAnimation(newAnimation)
  }

  render() {
    const { animation } = this.props 
    return (
      <div className='animation-container'>
        <div className='class-container'>
          <p className='class-selector'>.square <span>{'{'}</span></p>
          <div className='props-container'>
            <p className='ani-prop name'>name<span>:</span></p>
            <p className='prop-input name' id='name'>{animation.properties.name}</p>
          </div>
          {
            aniPropsLabels.map( (prop, i) => {
              return (
                <div className='props-container' key={uid(prop)}>
                  <p className={`ani-prop ${[aniProps[i]]}`}>{prop}<span>:</span></p>
                  <input className={`prop-input ${[aniProps[i]]}`} 
                    type='text' 
                    value={animation.properties[aniProps[i]]}
                    onChange={e => this.saveForm(e)}></input>
                </div>
              )
            })
          }
          <p className='close-curly'>{'}'}</p>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  animation: state.animation
})

export const mapDispatchToProps = (dispatch) => ({
  updateAnimation: (animation) => dispatch(loadAnimation(animation))
})

export default connect(mapStateToProps, mapDispatchToProps)(PropertiesEditor)
