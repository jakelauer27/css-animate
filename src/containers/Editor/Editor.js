import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Route, Switch, NavLink, withRouter, Link } from 'react-router-dom'
import KeyframesEditor from '../KeyframesEditor/KeyframesEditor'
import PropertiesEditor from '../PropertiesEditor/PropertiesEditor'
import { loadAnimation } from '../../actions/actions'
import * as CSSInsertion from '../../utils/keyframesInsertion'
import animationsData from '../../utils/data'
import CopyPopup from '../CopyPopup/CopyPopup'

const animationKeys = Object.keys(animationsData)

export class Editor extends Component {

  componentDidMount() {
    this.loadNewAnimation(this.props.currentAnimation)
  }

  loadNewAnimation(animationName) {
    const animationKey = animationKeys.find( key => key === animationName)
    const newAnimation = JSON.parse(JSON.stringify(animationsData[animationKey]))
    this.props.loadNewAnimation(newAnimation)
    CSSInsertion.updateKeyframes(newAnimation.keyframes)
  }
  
  resetInputValidation() {
    document.querySelectorAll('input').forEach( input => {
      input.classList.remove('red');
    });
    document.querySelector('.play-btn').removeAttribute('disabled');
  }


  render() {
    const { animation, currentAnimation } = this.props
    if (!animation.keyframes) {
      return <div></div>
    }
    return (
      <div className='editor-component'>
        <div className='editor-container'>
          <div className='editor-options'>
            <NavLink 
              to={`/${animation.keyframes.name}/properties`}
              className='animation-btn'
              activeClassName='selected'>properties
            </NavLink>
            <NavLink 
              to={`/${animation.keyframes.name}/keyframes`}
              className='keyframes-btn'
              activeClassName='selected'>keyframes
            </NavLink>
          </div>
          <div className='editor'>
          {
            <Switch>
              <Route path={`${process.env.PUBLIC_URL}/${animation.keyframes.name}/keyframes`} component={KeyframesEditor} />
              <Route path={`${process.env.PUBLIC_URL}/${animation.keyframes.name}/properties`} component={PropertiesEditor}/>
            </Switch>
          }
          </div>
        </div>
        <div className='editor-bottom-btns-container'>
          <Link to={`/${animation.keyframes.name}/properties/copy`} className='lower-btn copy-btn'>
            copy code   
          </Link>
          <Route path={`${process.env.PUBLIC_URL}/${animation.keyframes.name}/properties/copy`} render={() => <CopyPopup currentPage={currentAnimation}/>} />
          <button className={`lower-btn reset-btn`} onClick={() => {
              this.loadNewAnimation(this.props.currentAnimation)
              this.resetInputValidation()  
            }
            }>
            reset
          </button>
        </div>
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({
  animation: state.animation
})

export const mapDispatchToProps = (dispatch) => ({
  loadNewAnimation: (animation) => dispatch(loadAnimation(animation))
})

export default connect(mapStateToProps, mapDispatchToProps)(Editor)
