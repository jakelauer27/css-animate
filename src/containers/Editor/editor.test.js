import { Editor, mapStateToProps, mapDispatchToProps } from './Editor' 
import React from 'react'
import { shallow, mount } from 'enzyme'
import { loadAnimation } from '../../actions/actions'
import * as CSSInsertion from '../../utils/keyframesInsertion'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import KeyframesEditor from '../KeyframesEditor/KeyframesEditor'
import PropertiesEditor from '../PropertiesEditor/PropertiesEditor'
import CopyPopup from '../CopyPopup/CopyPopup'


jest.mock('../../utils/keyframesInsertion')

CSSInsertion.updateKeyframes.mockImplementation(() => jest.fn())

describe('Editor', () => {
  let wrapper
  const mockLoadAnimation = jest.fn()
  const mockAnimation = { 
    'properties': {
      'name': 'slideInX',
      'duration': '1.5s',
      'timingFunction': 'ease',
      'delay': '0s',
      'iterationCount': '1',
      'direction': 'normal',
      'fillMode': 'forwards'
    },
    'keyframes': {
      'name': 'slideInX',
      'sections': [
        {
          'name': '0%',
          'label': '0%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(-300px)'
            }
          ]
        },
        {
          'name': '100%',
          'label': '100%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(0px)'
            }
          ]
        },
      ]
    }
  }

  beforeEach(() => {
    wrapper = shallow(<Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if there is no animation', () => {
    wrapper = shallow(<Editor animation={{keyframes: null}} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call loadNewAnimation when reset button is clicked', () => {
    const loadNewSpy = jest.spyOn(wrapper.instance(), 'loadNewAnimation')
    wrapper.find('.reset-btn').simulate('click')
    expect(loadNewSpy).toHaveBeenCalled()
  })

  it('should call resetInputValidion when reset button is clicked', () => {
    const resestInputSpy = jest.spyOn(wrapper.instance(), 'resetInputValidation')
    wrapper.find('.reset-btn').simulate('click')
    expect(resestInputSpy).toHaveBeenCalled()
  })

  describe('componentDidMount', () => {

    it('should call loadNewAnimation', () => {
      wrapper = shallow(<Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>, {disableLifecycleMethods: true})
      const spy = jest.spyOn(wrapper.instance(), 'loadNewAnimation')
      wrapper.instance().componentDidMount()
      expect(spy).toHaveBeenCalledWith('slideInX')
    })
  })

  describe('loadNewAnimation', () => {
    wrapper = shallow(<Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>, {disableLifecycleMethods: true})

    it('should call loadNewAnimation from props', () => {
      const expected = {"keyframes": {"name": "slideInX", "sections": [{"label": "0%", "name": "0%", "properties": [{"name": "transform", "value": "translateX(-300px)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}, "properties": {"delay": "0s", "direction": "normal", "duration": "1.5s", "fillMode": "forwards", "iterationCount": "1", "name": "slideInX", "timingFunction": "ease"}}
      wrapper.instance().loadNewAnimation('slideInX')
      expect(mockLoadAnimation).toHaveBeenCalledWith(expected);
    })

    it('should call updateKeyframes', () => {
      const expected = {"name": "slideInX", "sections": [{"label": "0%", "name": "0%", "properties": [{"name": "transform", "value": "translateX(-300px)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}
      wrapper.instance().loadNewAnimation('slideInX')
      expect(CSSInsertion.updateKeyframes).toHaveBeenCalledWith(expected);
    })
  })

  describe('resetInputValidation', () => {
    document.querySelectorAll = jest.fn().mockImplementation(() => {
      return [{classList: {remove: jest.fn()}}]
    })
    document.querySelector = jest.fn().mockImplementation(() => ({removeAttribute: jest.fn()}))
  
    
    it('should remove red class from all inputs', () => {
      wrapper.instance().resetInputValidation()
      expect(document.querySelectorAll).toHaveBeenCalled()
    })

    it('should remove disabled attr from play btn', () => {
      wrapper.instance().resetInputValidation()
      expect(document.querySelector).toHaveBeenCalled()
    })
  })

  describe('routes', () => {

    it('should render keyframes component if on keyframes route', () => {
      wrapper = mount(
        <Provider store={createStore(() => ({animation: mockAnimation}))}>
          <MemoryRouter initialEntries={['/slideInX/keyframes']}>
            <Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>
          </MemoryRouter>
        </Provider>
        )
        expect(wrapper.find(KeyframesEditor)).toHaveLength(1)
    })
    
    it('should render properties component if on properties route', () => {
      wrapper = mount(
        <Provider store={createStore(() => ({animation: mockAnimation}))}>
          <MemoryRouter initialEntries={['/slideInX/properties']}>
            <Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>
          </MemoryRouter>
        </Provider>
        )
        expect(wrapper.find(PropertiesEditor)).toHaveLength(1)
    })

    it('should render copyCode component if on copyCode route', () => {
      wrapper = mount(
        <Provider store={createStore(() => ({animation: mockAnimation}))}>
          <MemoryRouter initialEntries={['/slideInX/properties/copy']}>
            <Editor animation={mockAnimation} loadNewAnimation={mockLoadAnimation} currentAnimation={'slideInX'}/>
          </MemoryRouter>
        </Provider>
        )
        expect(wrapper.find(CopyPopup)).toHaveLength(1)
    })
  })

  describe('mapStateToProps', () => {

    it('should return a props object with an animation property', () => {
      const mockState = {animation: mockAnimation}
      const expected = {animation: mockAnimation}
      expect(mapStateToProps(mockState)).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('should return a props object with a loadNewAnimation property', () => {
        const mockDispatch = jest.fn()
        const actionToDispatch = loadAnimation(mockAnimation)
        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.loadNewAnimation(mockAnimation)
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})