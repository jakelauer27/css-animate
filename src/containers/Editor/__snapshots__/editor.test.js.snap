// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor
    animation={
      Object {
        "keyframes": Object {
          "name": "slideInX",
          "sections": Array [
            Object {
              "label": "0%",
              "name": "0%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(-300px)",
                },
              ],
            },
            Object {
              "label": "100%",
              "name": "100%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(0px)",
                },
              ],
            },
          ],
        },
        "properties": Object {
          "delay": "0s",
          "direction": "normal",
          "duration": "1.5s",
          "fillMode": "forwards",
          "iterationCount": "1",
          "name": "slideInX",
          "timingFunction": "ease",
        },
      }
    }
    currentAnimation="slideInX"
    loadNewAnimation={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "keyframes": Object {
                "name": "slideInX",
                "sections": Array [
                  Object {
                    "label": "0%",
                    "name": "0%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(-300px)",
                      },
                    ],
                  },
                  Object {
                    "label": "100%",
                    "name": "100%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(0px)",
                      },
                    ],
                  },
                ],
              },
              "properties": Object {
                "delay": "0s",
                "direction": "normal",
                "duration": "1.5s",
                "fillMode": "forwards",
                "iterationCount": "1",
                "name": "slideInX",
                "timingFunction": "ease",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="editor-container"
        >
          <div
            className="editor-options"
          >
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="animation-btn"
              to="/slideInX/properties"
            >
              properties
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="keyframes-btn"
              to="/slideInX/keyframes"
            >
              keyframes
            </NavLink>
          </div>
          <div
            className="editor"
          >
            <Switch>
              <Route
                component={[Function]}
                path="/slideInX/keyframes"
              />
              <Route
                component={[Function]}
                path="/slideInX/properties"
              />
            </Switch>
          </div>
        </div>,
        <div
          className="editor-bottom-btns-container"
        >
          <Link
            className="lower-btn copy-btn"
            replace={false}
            to="/slideInX/properties/copy"
          >
            copy code
          </Link>
          <Route
            path="/slideInX/properties/copy"
            render={[Function]}
          />
          <button
            className="lower-btn reset-btn"
            onClick={[Function]}
          >
            reset
          </button>
        </div>,
      ],
      "className": "editor-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="editor-options"
            >
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="animation-btn"
                to="/slideInX/properties"
              >
                properties
              </NavLink>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="keyframes-btn"
                to="/slideInX/keyframes"
              >
                keyframes
              </NavLink>
            </div>,
            <div
              className="editor"
            >
              <Switch>
                <Route
                  component={[Function]}
                  path="/slideInX/keyframes"
                />
                <Route
                  component={[Function]}
                  path="/slideInX/properties"
                />
              </Switch>
            </div>,
          ],
          "className": "editor-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="animation-btn"
                  to="/slideInX/properties"
                >
                  properties
                </NavLink>,
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="keyframes-btn"
                  to="/slideInX/keyframes"
                >
                  keyframes
                </NavLink>,
              ],
              "className": "editor-options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected",
                  "aria-current": "page",
                  "children": "properties",
                  "className": "animation-btn",
                  "to": "/slideInX/properties",
                },
                "ref": null,
                "rendered": "properties",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected",
                  "aria-current": "page",
                  "children": "keyframes",
                  "className": "keyframes-btn",
                  "to": "/slideInX/keyframes",
                },
                "ref": null,
                "rendered": "keyframes",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Switch>
                <Route
                  component={[Function]}
                  path="/slideInX/keyframes"
                />
                <Route
                  component={[Function]}
                  path="/slideInX/properties"
                />
              </Switch>,
              "className": "editor",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    path="/slideInX/keyframes"
                  />,
                  <Route
                    component={[Function]}
                    path="/slideInX/properties"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/slideInX/keyframes",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/slideInX/properties",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="lower-btn copy-btn"
              replace={false}
              to="/slideInX/properties/copy"
            >
              copy code
            </Link>,
            <Route
              path="/slideInX/properties/copy"
              render={[Function]}
            />,
            <button
              className="lower-btn reset-btn"
              onClick={[Function]}
            >
              reset
            </button>,
          ],
          "className": "editor-bottom-btns-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "copy code",
              "className": "lower-btn copy-btn",
              "replace": false,
              "to": "/slideInX/properties/copy",
            },
            "ref": null,
            "rendered": "copy code",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/slideInX/properties/copy",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "reset",
              "className": "lower-btn reset-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "reset",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="editor-container"
          >
            <div
              className="editor-options"
            >
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="animation-btn"
                to="/slideInX/properties"
              >
                properties
              </NavLink>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="keyframes-btn"
                to="/slideInX/keyframes"
              >
                keyframes
              </NavLink>
            </div>
            <div
              className="editor"
            >
              <Switch>
                <Route
                  component={[Function]}
                  path="/slideInX/keyframes"
                />
                <Route
                  component={[Function]}
                  path="/slideInX/properties"
                />
              </Switch>
            </div>
          </div>,
          <div
            className="editor-bottom-btns-container"
          >
            <Link
              className="lower-btn copy-btn"
              replace={false}
              to="/slideInX/properties/copy"
            >
              copy code
            </Link>
            <Route
              path="/slideInX/properties/copy"
              render={[Function]}
            />
            <button
              className="lower-btn reset-btn"
              onClick={[Function]}
            >
              reset
            </button>
          </div>,
        ],
        "className": "editor-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="editor-options"
              >
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="animation-btn"
                  to="/slideInX/properties"
                >
                  properties
                </NavLink>
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="keyframes-btn"
                  to="/slideInX/keyframes"
                >
                  keyframes
                </NavLink>
              </div>,
              <div
                className="editor"
              >
                <Switch>
                  <Route
                    component={[Function]}
                    path="/slideInX/keyframes"
                  />
                  <Route
                    component={[Function]}
                    path="/slideInX/properties"
                  />
                </Switch>
              </div>,
            ],
            "className": "editor-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="selected"
                    aria-current="page"
                    className="animation-btn"
                    to="/slideInX/properties"
                  >
                    properties
                  </NavLink>,
                  <NavLink
                    activeClassName="selected"
                    aria-current="page"
                    className="keyframes-btn"
                    to="/slideInX/keyframes"
                  >
                    keyframes
                  </NavLink>,
                ],
                "className": "editor-options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "selected",
                    "aria-current": "page",
                    "children": "properties",
                    "className": "animation-btn",
                    "to": "/slideInX/properties",
                  },
                  "ref": null,
                  "rendered": "properties",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "selected",
                    "aria-current": "page",
                    "children": "keyframes",
                    "className": "keyframes-btn",
                    "to": "/slideInX/keyframes",
                  },
                  "ref": null,
                  "rendered": "keyframes",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Switch>
                  <Route
                    component={[Function]}
                    path="/slideInX/keyframes"
                  />
                  <Route
                    component={[Function]}
                    path="/slideInX/properties"
                  />
                </Switch>,
                "className": "editor",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      path="/slideInX/keyframes"
                    />,
                    <Route
                      component={[Function]}
                      path="/slideInX/properties"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/slideInX/keyframes",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/slideInX/properties",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="lower-btn copy-btn"
                replace={false}
                to="/slideInX/properties/copy"
              >
                copy code
              </Link>,
              <Route
                path="/slideInX/properties/copy"
                render={[Function]}
              />,
              <button
                className="lower-btn reset-btn"
                onClick={[Function]}
              >
                reset
              </button>,
            ],
            "className": "editor-bottom-btns-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "copy code",
                "className": "lower-btn copy-btn",
                "replace": false,
                "to": "/slideInX/properties/copy",
              },
              "ref": null,
              "rendered": "copy code",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/slideInX/properties/copy",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "reset",
                "className": "lower-btn reset-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "reset",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Editor should match the snapshot if there is no animation 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor
    animation={
      Object {
        "keyframes": null,
      }
    }
    currentAnimation="slideInX"
    loadNewAnimation={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "keyframes": Object {
                "name": "slideInX",
                "sections": Array [
                  Object {
                    "label": "0%",
                    "name": "0%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(-300px)",
                      },
                    ],
                  },
                  Object {
                    "label": "100%",
                    "name": "100%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(0px)",
                      },
                    ],
                  },
                ],
              },
              "properties": Object {
                "delay": "0s",
                "direction": "normal",
                "duration": "1.5s",
                "fillMode": "forwards",
                "iterationCount": "1",
                "name": "slideInX",
                "timingFunction": "ease",
              },
            },
          ],
          Array [
            Object {
              "keyframes": Object {
                "name": "slideInX",
                "sections": Array [
                  Object {
                    "label": "0%",
                    "name": "0%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(-300px)",
                      },
                    ],
                  },
                  Object {
                    "label": "100%",
                    "name": "100%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(0px)",
                      },
                    ],
                  },
                ],
              },
              "properties": Object {
                "delay": "0s",
                "direction": "normal",
                "duration": "1.5s",
                "fillMode": "forwards",
                "iterationCount": "1",
                "name": "slideInX",
                "timingFunction": "ease",
              },
            },
          ],
          Array [
            Object {
              "keyframes": Object {
                "name": "slideInX",
                "sections": Array [
                  Object {
                    "label": "0%",
                    "name": "0%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(-300px)",
                      },
                    ],
                  },
                  Object {
                    "label": "100%",
                    "name": "100%",
                    "properties": Array [
                      Object {
                        "name": "transform",
                        "value": "translateX(0px)",
                      },
                    ],
                  },
                ],
              },
              "properties": Object {
                "delay": "0s",
                "direction": "normal",
                "duration": "1.5s",
                "fillMode": "forwards",
                "iterationCount": "1",
                "name": "slideInX",
                "timingFunction": "ease",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
